gax-java/gax/src/main/java/com/google/api/gax/util/TimeConversionUtils.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are
+ * met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ * notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above
+ * copyright notice, this list of conditions and the following disclaimer
+ * in the documentation and/or other materials provided with the
+ * distribution.
+ *     * Neither the name of Google LLC nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package com.google.api.gax.util;
+
+public class TimeConversionUtils {
+  public static java.time.Duration toJavaTimeDuration(org.threeten.bp.Duration source) {
+    if (source == null) {
+      return null;
+    }
+    return java.time.Duration.ofNanos(source.toNanos());
+  }
+
+  public static org.threeten.bp.Duration toThreetenDuration(java.time.Duration source) {
+    if (source == null) {
+      return null;
+    }
+    return org.threeten.bp.Duration.ofNanos(source.toNanos());
+  }
+
+  public static java.time.Instant toJavaTimeInstant(org.threeten.bp.Instant source) {
+    if (source == null) {
+      return null;
+    }
+    return java.time.Instant.ofEpochMilli(source.toEpochMilli());
```

Comment:
FWIW, this truncates sub-millisecond precision. I think what you probably want to do is instead:

`return java.time.Instant.ofEpochSecond(source.getEpochSecond(), source.getNano());`

Reply:
CC: @diegomarquezp Can you take a look at this?

Reply:
I confirmed this causes trouble:

```java
  @Test
  void testToThreeteenTimeInstant_bigInput_doesNotTruncate() {
    java.time.Instant jtInstant = java.time.Instant.MAX;
    org.threeten.bp.Instant ttInstant = TimeConversionUtils.toThreetenInstant(jtInstant);
    assertEquals(jtInstant.toEpochMilli(), ttInstant.toEpochMilli());
  }
```

It did not truncate, though. It failed with an overflow exception:

```
java.lang.ArithmeticException: long overflow

	at java.base/java.lang.Math.multiplyExact(Math.java:1033)
	at java.base/java.lang.Math.multiplyExact(Math.java:1009)
	at java.base/java.time.Instant.toEpochMilli(Instant.java:1262)
	at com.google.api.gax.util.TimeConversionUtils.toThreetenInstant(TimeConversionUtils.java:59)
```

I'll leave this test as a standard for a coming PR.